pkg/generic/remove_options.go

```diff
@@ -0,0 +1,17 @@
+package generic
+
+import "k8s.io/apimachinery/pkg/runtime"
+
+// OnRemoveOption allows configuring OnRemove handlers
+type OnRemoveOption func(*objectLifecycleAdapter)
+
+// WithCondition will make OnRemove handlers to ignore resources not matching the provided condition
+func WithCondition(f func(runtime.Object) bool) OnRemoveOption {
+	return func(o *objectLifecycleAdapter) {
+		o.condition = f
```

Comment:
You might want to clarify that the latest `WithCondition` wins in case multiple are specified (one could think they are combined with AND)

Reply:
Thanks, amended the comment at 86560d9